src/ws/cockpithandlers.c

```diff
@@ -341,6 +341,28 @@ build_environment (GHashTable *os_release)
   if (ca_path)
     json_object_set_string_member (object, "CACertUrl", "/ca.cer");
 
+  gchar *contents = NULL;
+  gsize len;
+  GError *error = NULL;
+
+  const gchar *banner = cockpit_conf_string ("Session", "Banner");
+  if (banner)
+    {
+      // TODO: parse macros (see `man agetty` for possible macros)
+      g_file_get_contents (banner, &contents, &len, &error);
+    }
+
+  if (error)
+    {
+      g_message ("error loading contents of banner: %s", error->message);
+      g_clear_error (&error);
+    }
+  else
+    {
+      json_object_set_string_member (object, "etcIssue", contents);
```

Comment:
Please rename the key to `banner`, as it's not technically tied to /etc/issue any more.